swift-ci/master/centos/7/Dockerfile

```diff
@@ -48,6 +48,38 @@ RUN ldconfig
 
 RUN sed -i -e 's/\*__block/\*__libc_block/g' /usr/include/unistd.h
 
+ARG SWIFT_SIGNING_KEY=A62AE125BBBFBB96A6E042EC925CC1CCED3D1561
+ARG SWIFT_PLATFORM=centos7
+ARG SWIFT_BRANCH=swift-5.8.1-release
+ARG SWIFT_VERSION=swift-5.8.1-RELEASE
+ARG SWIFT_WEBROOT=https://download.swift.org
+
+ENV SWIFT_SIGNING_KEY=$SWIFT_SIGNING_KEY \
+    SWIFT_PLATFORM=$SWIFT_PLATFORM \
+    SWIFT_BRANCH=$SWIFT_BRANCH \
+    SWIFT_VERSION=$SWIFT_VERSION \
+    SWIFT_WEBROOT=$SWIFT_WEBROOT
+
+RUN set -e; \
+    SWIFT_WEBDIR="$SWIFT_WEBROOT/$SWIFT_BRANCH/$(echo $SWIFT_PLATFORM | tr -d .)" \
+    && SWIFT_BIN_URL="$SWIFT_WEBDIR/$SWIFT_VERSION/$SWIFT_VERSION-$SWIFT_PLATFORM.tar.gz" \
+    && SWIFT_SIG_URL="$SWIFT_BIN_URL.sig" \
+    # - Download the GPG keys, Swift toolchain, and toolchain signature, and verify.
+    && export GNUPGHOME="$(mktemp -d)" \
+    && curl -fsSL "$SWIFT_BIN_URL" -o swift.tar.gz "$SWIFT_SIG_URL" -o swift.tar.gz.sig \
+    && gpg --batch --quiet --keyserver keyserver.ubuntu.com --recv-keys "$SWIFT_SIGNING_KEY" \
+    && gpg --batch --verify swift.tar.gz.sig swift.tar.gz \
+    && mkdir -p /opt/swift \
+    # - Unpack the toolchain, set libs permissions, and clean up.
+    && tar -xzf swift.tar.gz --directory /opt/swift/ --strip-components=1 \
+    && chmod -R o+r /opt/swift/usr/lib/swift \
+    && ln -s /opt/swift/usr/lib/swift /usr/lib/swift \
+    && rm -rf "$GNUPGHOME" swift.tar.gz.sig swift.tar.gz
+
+ENV PATH="${PATH}:/opt/swift/usr/bin"
+
+RUN swift --version
```

Comment:
this is a surprising change. why do we need to do this? should we fix the docker image instead if this is fixing something?

Reply:
Ideally *all* our release images would install into `/opt/swift/<version>` instead of `/usr`. But we don't want to break current clients, so for now we're just updating the builder image.

Without this change, Swift uses the toolchain-clang to build and then uses the system C++ stdlib instead of devtoolset as it should. The system stdlib is wildly out of date (unsurprising) and thus doesn't have any new C++ features.

@shahmishal given all the other images are using the toolchain clang, perhaps we should make this `ENV PATH=/opt/swift/usr/bin:${PATH}` instead of appending?

Reply:
> @shahmishal given all the other images are using the toolchain clang, perhaps we should make this ENV PATH=/opt/swift/usr/bin:${PATH} instead of appending?

Yes, that make sense.

Reply:
> this is a surprising change. why do we need to do this? should we fix the docker image instead if this is fixing something?

We can change this in 5.9 release, if the community is ok with the change. I would like to avoid this type of change in dot release.